vault/resource_database_secret_backend_connection_test.go

```diff
@@ -767,6 +767,54 @@ func TestAccDatabaseSecretBackendConnection_postgresql(t *testing.T) {
 	})
 }
 
+// This test makes sure that the DB connection resource is still
+// operational even when an external rotate root call is made to update
+// its credentials
+func TestAccDatabaseSecretBackendConnection_externalRotateRoot(t *testing.T) {
+	MaybeSkipDBTests(t, dbEnginePostgres)
+
+	username := "postgres"
+	password := "NotSecurePassword1"
+	backend := acctest.RandomWithPrefix("tf-test-db")
+	pluginName := dbEnginePostgres.DefaultPluginName()
+	name := acctest.RandomWithPrefix("db")
+	maxOpenConnections := "16"
+	updatedMaxOpenConnections := "20"
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: providerFactories,
+		PreCheck:          func() { testutil.TestAccPreCheck(t) },
+		CheckDestroy:      testAccDatabaseSecretBackendConnectionCheckDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatabaseSecretBackendConnectionConfig_PostgresRotateRoot(name, backend, username, password, maxOpenConnections),
+				Check: testComposeCheckFuncCommonDatabaseSecretBackend(name, backend, pluginName,
+					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "postgresql.0.username", username),
+					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "postgresql.0.password", password),
+				),
+			},
+			{
+				PreConfig: func() {
+					client := testProvider.Meta().(*provider.ProviderMeta).GetClient()
+					rotateRootPath := fmt.Sprintf("%s/rotate-root/%s", backend, name)
+					resp, err := client.Logical().Write(rotateRootPath, nil)
```

Comment:
The rotate root does not seem to be working as expected. At present this test passes even when it should fail since no root rotation is actually taking place. Looking into this a bit more